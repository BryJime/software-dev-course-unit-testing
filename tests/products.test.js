const { calculateDiscount, filterProducts, sampleInventory, sortInventory } = require('../products.js');


describe("The shopping cart", () => {

    test("should calculate discout accuratley", () => {
        expect(calculateDiscount(100, .10)).toEqual(90)
    })

    test("should account for negative numbers", () => {
        expect(calculateDiscount(100, .10)).not.toEqual(-90);
    })


    test("should account for discount over than 1", () => {
        expect(calculateDiscount(100, 1.2)).toEqual(null);
    })

    test("should filter to find yoga mat", () => {
        expect(filterProducts(sampleInventory, (products) => products.name === "Yoga Mat")).toEqual([
  {
    id: 9,
    name: 'Yoga Mat',
    category: 'Fitness',
    price: 35,
    inStock: true,
    rating: 4.7
  }
]);
    })

    test("should return empty array if callback is not a function", () => {
        expect(filterProducts(sampleInventory, 'string')).toEqual([]);
    })

    test("should return empty array if function is broken", () => {
        expect(filterProducts(sampleInventory, () => {})).toEqual([])
    })

    test("should sort item names in alphbetical order", () => {
        expect(sortInventory(sampleInventory, "name" )).toMatchSnapshot();
    })

    test("should return empty array if inventory is not an array", () => {
        expect(sortInventory("inventory", "name")).toEqual([]);
    })

    test("should return 0 if key is not found in array", () => {
        expect(sortInventory(sampleInventory, undefined)).toEqual([])
    })

});










