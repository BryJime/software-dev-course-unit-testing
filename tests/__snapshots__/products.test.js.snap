// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The shopping cart should sort item names in alphbetical order 1`] = `
[
  {
    "category": "Electronics",
    "id": 7,
    "inStock": true,
    "name": "Bluetooth Headset",
    "price": 59.99,
    "rating": 4.6,
  },
  {
    "category": "Home Goods",
    "id": 4,
    "inStock": true,
    "name": "Ceramic Mug",
    "price": 12,
    "rating": 4,
  },
  {
    "category": "Apparel",
    "id": 1,
    "inStock": true,
    "name": "Classic Tee",
    "price": 19.99,
    "rating": 4.2,
  },
  {
    "category": "Apparel",
    "id": 3,
    "inStock": true,
    "name": "Cozy Hoodie",
    "price": 49.5,
    "rating": 4.5,
  },
  {
    "category": "Home Goods",
    "id": 10,
    "inStock": true,
    "name": "Desk Lamp",
    "price": 22.5,
    "rating": 4.4,
  },
  {
    "category": "Books",
    "id": 5,
    "inStock": false,
    "name": "Hardcover Book",
    "price": 24.99,
    "rating": 4.9,
  },
  {
    "category": "Footwear",
    "id": 2,
    "inStock": false,
    "name": "Running Sneakers",
    "price": 89.95,
    "rating": 4.8,
  },
  {
    "category": "Outdoor",
    "id": 8,
    "inStock": false,
    "name": "Water Bottle",
    "price": 15,
    "rating": 4.1,
  },
  {
    "category": "Electronics",
    "id": 6,
    "inStock": true,
    "name": "Wireless Mouse",
    "price": 29.99,
    "rating": 4.3,
  },
  {
    "category": "Fitness",
    "id": 9,
    "inStock": true,
    "name": "Yoga Mat",
    "price": 35,
    "rating": 4.7,
  },
]
`;
